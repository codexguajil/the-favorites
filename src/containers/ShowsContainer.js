import React, { Component } from 'react';
import Movie from '../components/Movie';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

class ShowsContainer extends Component{
  renderShows = () => {
    return this.props.shows.map(show => {
      return (
        <Link to={`/shows/${show.id}`} key={show.id} >
          <Movie  
            {...show}
          />
        </Link>
      )
    })
  }

  render() {
    return (
      <div className="shows-container">
        {this.renderShows()}
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  shows: state.shows
})

export default connect(mapStateToProps)(ShowsContainer)