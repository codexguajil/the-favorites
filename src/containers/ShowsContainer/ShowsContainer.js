import React, { Component } from 'react';
import Movie from '../Movie/Movie';
import { connect } from 'react-redux';
import propTypes from 'prop-types';

class ShowsContainer extends Component{
  renderShows = () => {
    return this.props.shows.map(show => {
      return (
        <Movie type='shows' 
          {...show}
        />
      )
    })
  }

  render() {
    return (
      <div className="shows-container">
        {this.renderShows()}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  shows: state.shows
})

export default connect(mapStateToProps)(ShowsContainer)

ShowsContainer.propTypes = {
  shows: propTypes.array.isRequired
}