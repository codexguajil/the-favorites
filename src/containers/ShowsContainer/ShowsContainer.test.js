import React from 'react';
import ShowsContainer from './ShowsContainer'
import { mapStateToProps } from './ShowsContainer'
import { shallow } from 'enzyme'

describe('ShowsContainer', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <ShowsContainer />
    )

    expect(wrapper).toMatchSnapshot()
  });
  
  describe('mapStateToProps', () => {
    it.skip('should show previous value for state favorites', () => {
      const initialState = {
        movies: [{}, {}, {}]
      }

      expect(mapStateToProps(initialState).favorites).toEqual([{}, {}, {}])
    });
  })
})