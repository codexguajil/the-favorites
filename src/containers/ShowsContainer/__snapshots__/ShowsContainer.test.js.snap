// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowsContainer should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
